src/index.js

```diff
@@ -26,8 +26,17 @@ export default function(url, prev) {
   // https://github.com/Updater/node-sass-json-importer/issues/21
   delete require.cache[require.resolve(file)];
 
+  var contents = require(file);
+  try {
+    if (/\.js$/.test(url)) {
+      contents = JSON.parse(JSON.stringify(contents));
+    }
+  } catch(e) {
+    return sass.NULL;
```

Comment:
Sorry, didn't realize you covered it with the "importing non-valid JSON from JS exports" test. I forgot that `node-sass` would throw if we ignored the file since it would try to process it itself. 

One way to more clearly communicate the expectation of this test would be to write a test-only JS importer, apply it as the second importer to the JSON importer and have the expectation be that it gets executed (since the JSON importer is ignored). But that's a lot more work and not strictly necessary, just extra credit ;). 

I think we're good as long as the test description communicates that we're skipping JS files that aren't exporting JSON.